src/test/java/org/javarosa/core/model/DynamicSelectUpdateTest.java

```diff
@@ -0,0 +1,323 @@
+            head(
+                title("Multilingual dynamic select"),
+                model(
+                    t("itext",
+                        t("translation lang=\"fr\"",
+                            t("text id=\"choices-0\"",
+                                t("value", "A (fr)")),
+                            t("text id=\"choices-1\"",
+                                t("value", "B (fr)")),
+                            t("text id=\"choices-2\"",
+                                t("value", "C (fr)"))
+                        ),
+                        t("translation lang=\"en\"",
+                            t("text id=\"choices-0\"",
+                                t("value", "A (en)")),
+                            t("text id=\"choices-1\"",
+                                t("value", "B (en)")),
+                            t("text id=\"choices-2\"",
+                                t("value", "C (en)"))
+                        )),
+                    mainInstance(
+                        t("data id=\"multilingual-select\"",
+                            t("select"))),
+
+                    instance("choices",
+                        t("item", t("itextId", "choices-0"), t("name", "a")),
+                        t("item", t("itextId", "choices-1"), t("name", "b")),
+                        t("item", t("itextId", "choices-2"), t("name", "c"))))),
+            body(
+                select1Dynamic("/data/select", "instance('choices')/root/item", "name", "jr:itext(itextId)"))
+            ));
+
+        scenario.setLanguage("en");
+        assertThat(scenario.choicesOf("/data/select").size(), is(3));
+        assertThat(scenario.choicesOf("/data/select"), containsInAnyOrder(
+            choice("a", "choices-0"),
+            choice("b", "choices-1"),
+            choice("c", "choices-2")));
+    }
+    //endregion
+
+    //region Caching
+    @Test
+    public void selectWithUnchangedTriggers_providesCachedChoiceList() throws Exception {
+        Scenario scenario = Scenario.init("Select", html(
+            head(
+                title("Select"),
+                model(
+                    mainInstance(
+                        t("data id=\"select\"",
+                            t("filter"),
+                            t("select"))),
+
+                    instance("choices",
+                        item("aa", "A"),
+                        item("aaa", "AA"),
+                        item("bb", "B"),
+                        item("bbb", "BB"))),
+                body(
+                    input("/data/filter"),
+                    select1Dynamic("/data/select", "instance('choices')/root/item[starts-with(value,/data/filter)]")
+                )
+            )));
+
+        scenario.answer("/data/filter", "a");
+        List<SelectChoice> choices = scenario.choicesOf("/data/select");
+        assertThat(scenario.choicesOf("/data/select"), sameInstance(choices));
```

Comment:
This actually works! Nice.

Reply:
Yeah, seems like a reasonable way to test for the behavior we want, right?!